### DISCUSSION TITLE
Solving TSP with a path planning algorithm

### DISCUSSION THREAD

**Discussion by surmitbhui** [google/or-tools#3114] [Routing (and legacy CP) questions]
Hi, 
I am new to using OR Tools and I am looking to solve a Travelling Salesman Problem with multiple obstacles defined between the desired routing waypoints. 
Currently, I tried the TSP example given on the [reference guide](https://developers.google.com/optimization/routing/tsp) but it doesn't seem to take into account any obstacles that it might encounter in deciding to move from one node to another. Is there any way to solve this issue using OR Tools and incorporating some sort of a path planning algorithm like A* with it? 

Thanks in advance.

**Comment by lperron** [comment]
the solver's input is a distance matrix (implicit or explicit). Infeasible arcs will be marked by using a very large value.
There is no path planning part, and the solver will struggle if finding a path between mandatory nodes is hard.

In other word, it is not tuned at all to do path planning.

If the problem is not too large, you can try the CP-SAT solver and look at example/python/tsp_sat.py

**Reply by surmitbhui** [reply]
Thank You. for your response. I'll look at the CP-SAT solver example for this case.
